linker/Linker.Steps/MarkStep.cs

```diff
@@ -1701,6 +1785,29 @@ protected virtual void DoAdditionalMethodProcessing (MethodDefinition method)
 		{
 		}
 
+		protected virtual bool ShouldMarkAsInstancePossible (MethodDefinition method)
+		{
+			// We don't need to mark it multiple times
+			if (Annotations.IsInstantiated (method.DeclaringType))
+				return false;
+
+			if (method.IsConstructor && !method.IsStatic)
+				return true;
+
+			if (method.DeclaringType.IsInterface)
+				return true;
+
+			return false;
+		}
+
+		protected virtual void MarkRequirementsForInstantiatedTypes (TypeDefinition type)
```

Comment:
Call it based on what it actually does, it should be something like `MarkInterfaceImplementation`

Reply:
It will be doing more soon.  I add to this in my base type sweeping branch that is coming.